src/app/features/stacks-transaction-request/hooks/use-stacks-transaction-summary.ts

```diff
@@ -72,13 +73,21 @@ export function useStacksTransactionSummary(token: CryptoCurrencies) {
     };
   }
 
+  function getSip10MemoDisplayText(tx: StacksTransaction): string {
+    const payload = tx.payload as ContractCallPayload;
+    const noMemoText = 'No memo';
+    if (!isDefined(payload.functionArgs[3])) {
+      return noMemoText;
+    }
+    const isSome = payload.functionArgs[3].type === ClarityType.OptionalSome;
+    return isSome ? bytesToUtf8(serializeCV(payload.functionArgs[3])) : noMemoText;
+  }
+
   function formSip10TxSummary(tx: StacksTransaction, symbol: string, decimals = 6) {
     const payload = tx.payload as ContractCallPayload;
     const fee = tx.auth.spendingCondition.fee;
     const txValue = Number((payload.functionArgs[0] as IntCV).value);
-    const isSome = payload.functionArgs[3].type === ClarityType.OptionalSome;
```

Comment:
In https://github.com/leather-wallet/extension/issues/5436 we can't send some SIP-10 tokens as the UI crashes on a missing `payload.functionArgs[3].type`. I checked and these tokens (DOGE) doesn't have any entry in `payload.functionArgs[3]`. 

I tried to add some code to set the memo to `No memo` if `payload.functionArgs[3]` is not present but the transaction fails to broadcast on the next step with a [broadcast error](https://github.com/leather-wallet/extension/issues/5436#issuecomment-2128553613)


It gives a `400 (Bad Request)` from https://api.hiro.so/v2/transactions then a transaction broadcast error:
```
{
    "error": "transaction rejected",
    "reason": "BadFunctionArgument",
    "reason_data": {
        "message": "IncorrectArgumentCount(4, 3)"
    },
    "txid": "33903b1126f0dea7234a58ec879f00cc1d38c844d26837024b01d76ace168cef"
}
```

Reply:
not sure here. 
I feel like there might be differences in contract implementations of sip10 tokens? that's why there are difference in `functionArgs`?